includes/class-newspack-popups-model.php

```diff
@@ -12,6 +12,87 @@
  */
 final class Newspack_Popups_Model {
 
+	/**
+	 * Retrieve all Popus (first 100).
+	 *
+	 * @return array Array of Popup objects.
+	 */
+	public static function retrieve_popups() {
+		$args = [
+			'post_type'      => Newspack_Popups::NEWSPACK_PLUGINS_CPT,
+			'post_status'    => 'publish',
+			'posts_per_page' => 100,
+		];
+
+		$popups = self::retrieve_popup_with_query( new WP_Query( $args ), true );
+
+		// Hacky check if this is a single popup object as opposed to an array of Pop-ups. TODO: replace.
+		if ( isset( $popups['id'] ) ) {
+			$popups = [ $popups ];
+		}
+		foreach ( $popups as &$popup ) {
```

Comment:
If I'm reading things correctly, it looks like `retrieve_popup_with_query` will return `null` if no popups exist. If so, this should not try and `foreach ( null )` because it will throw a notice. 

You could just put a `if ( empty( $popups ) ) { return []; }` somewhere above and it won't be an issue.

Reply:
I reworked this a bit in https://github.com/Automattic/newspack-popups/pull/34/commits/a7ca443afb45b858255fdfccb81b4348e60ab376 so that `retrieve_popup_with_query()` will always return an array.